### PR TITLE
Candidate card

### PR DESCRIPTION
**Pull request by vchepurnaya**
- add status bar
- fix bugs in candidate card
- add calendar in candidates

### CODE REVIEW

File: src/api/api.ts [hunk 1]
```diff
@@ -149,37 +149,42 @@ export const eventsApi = {
   },
 };
 
-const getAllCandidates = (params: PageParams) => {
-  return instance.get<TCandidate[]>('/candidate', { params });
+export const getCandidateIdByEmpolee = (id: number) => {
+  return instance.get<TInterviewTime[]>(`interviewtime?search=empId==${id}`).then(response => response.data);
```
Comment (ZmicerBoksha) [inline src/api/api.ts]:
```suggestion
  return instance.get<TInterviewTime[]>(`interviewtime?search=empId==${id}`).then({ data } => data);
```

File: src/api/api.ts [hunk 2]
```diff
@@ -149,37 +149,42 @@ export const eventsApi = {
...
 };
 
-const getCandidate = (id: number) => {
-  return instance.get(`/candidate/${id}`);
+export const getCandidateById = (candidateIds: number[]) => {
+  return instance.get<TCandidate[]>(`candidate?search=id=in=(${candidateIds.join()})`).then(response => response.data);
```
Comment (ZmicerBoksha) [inline src/api/api.ts]:
```suggestion
  return instance.get<TCandidate[]>(`candidate?search=id=in=(${candidateIds.join()})`).then({ data } => data);
```

File: src/api/api.ts [hunk 3]
```diff
@@ -149,37 +149,42 @@ export const eventsApi = {
...
 };
 
-const updateCandidate = (id: number, data: TCandidate) => {
-  return instance.put(`/candidate/${id}`, data);
+export const getStatusCandidateById = (candidateIds: number[]) => {
+  return instance
+    .get(`status/history/all?search=candidate.id=in=(${candidateIds.join()})`)
+    .then(response => response.data);
```
Comment (ZmicerBoksha) [inline src/api/api.ts]:
```suggestion
    .then({ data } => data);
```

File: src/components/candidates/candidate_card/candidate_interview/candidate_interview.tsx [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+      .then(({ data: { content } }) => {
+        setTs(content);
+      })
+      .catch(err => {
+        console.log(err);
+      });
+  }, []);
+
+  return (
+    <>
```
Comment (ZmicerBoksha) [inline src/components/candidates/candidate_card/candidate_interview/candidate_interview.tsx]: Do we need React.Fragment here?

File: src/components/candidates/candidate_card/candidate_progress/candidate_progress.tsx [hunk 5]
```diff
@@ -61,69 +64,106 @@ const useStyles = makeStyles((theme: Theme) =>
...
+}) => {
+  const [status, setStatus] = useState<TStatus[]>([]);
   const classes = useStyles();
-  const [activeStep, setActiveStep] = React.useState(0);
   const steps = getSteps();
+  const { activeStep } = useContext(Context);
+  const [currentStep, setCurrentStep] = useState<number>(1);
+
+  function getSteps() {
+    return status.map((item: any) => item.name);
```
Comment (ZmicerBoksha) [inline src/components/candidates/candidate_card/candidate_progress/candidate_progress.tsx]: no `any` please

File: src/components/candidates/candidate_card/candidate_progress/candidate_progress.tsx [hunk 6]
```diff
@@ -61,69 +64,106 @@ const useStyles = makeStyles((theme: Theme) =>
...
+  }
+
+  function getStepContent(stepIndex: number) {
+    return status.map((item: any, index: number) => {
```
Comment (ZmicerBoksha) [inline src/components/candidates/candidate_card/candidate_progress/candidate_progress.tsx]: no `any` please

File: src/components/candidates/candidate_card/candidate_progress/candidate_progress.tsx [hunk 7]
```diff
@@ -61,69 +64,106 @@ const useStyles = makeStyles((theme: Theme) =>
...
-    setActiveStep(prevActiveStep => prevActiveStep + 1);
+  const setCandidateStatus = (statusName: string) => {
+    editCandidateData({ status: statusName });
+    setCandidatesStatusColor(statusName);
   };
 
+  useEffect(() => {
+    statusServer
+      .getAllStatus()
+      .then(({ data }) => {
```
Comment (ZmicerBoksha) [inline src/components/candidates/candidate_card/candidate_progress/candidate_progress.tsx]: `.then(({ data: { content } }) => {
        setStatus(content);
      })`

File: src/components/candidates/candidate_card/candidate_schedule/candidate_schedule.tsx [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+      <Calendar
+        selectable={true}
+        views={['week']}
+        defaultView={'week'}
+        onNavigate={e => console.log('надо сохранить если мы переключаем неделю')}
+        localizer={localizer}
+        events={listOfEvents}
+        startAccessor="start"
+        onSelectEvent={event => console.log(event)}
+        onSelectSlot={e => handleSelect(e)}
```
Comment (ZmicerBoksha) [inline src/components/candidates/candidate_card/candidate_schedule/candidate_schedule.tsx]:
```suggestion
        onSelectSlot={handleSelect}
```

File: src/components/candidates/candidate_card/candidate_skills/candidate_skills.tsx [hunk 9]
```diff
@@ -50,8 +54,9 @@ const CandidateSkills: React.FC<TSoftSkillsProps> = ({
           </Button>
           <Button
             onClick={() => {
-              handleSubmitReview();
+              handleSubmitReview && handleSubmitReview();
```
Comment (ZmicerBoksha) [inline src/components/candidates/candidate_card/candidate_skills/candidate_skills.tsx]:
```suggestion
              handleSubmitReview?.();
```

File: src/components/staff/candidateTrello/candidateTrello.tsx [hunk 10]
```diff
@@ -1,81 +1,63 @@
...
     getData();
   }, []);
 
   let mass: any[] = [];
 
-  intreviewTimeData?.forEach((item) => {
+  intreviewTimeData?.forEach(item => {
     mass.push({
-      ["id"]: item.cnId, ["interviewDate"]: item.beginDate
+      ['id']: item.cnId,
```
Comment (ZmicerBoksha) [inline src/components/staff/candidateTrello/candidateTrello.tsx]:
```suggestion
      id: item.cnId,
```

File: src/components/staff/candidateTrello/candidateTrello.tsx [hunk 11]
```diff
@@ -1,81 +1,63 @@
...
+      ['interviewDate']: item.beginDate,
```
Comment (ZmicerBoksha) [inline src/components/staff/candidateTrello/candidateTrello.tsx]:
```suggestion
      interviewDate: item.beginDate,
```

File: src/components/staff/candidateTrello/candidateTrello.tsx [hunk 12]
```diff
@@ -1,81 +1,63 @@
...
-    mass[index]["primaryTechnology"] = candidatetsData[index].mainSkill;
+    mass[index]['firstName'] = candidatetsData[index].firstName;
+    mass[index]['lastName'] = candidatetsData[index].lastName;
+    mass[index]['primaryTechnology'] = candidatetsData[index].mainSkill;
   });
 
-
   statusHistoryData.forEach((item, index) => {
-    mass[index]["status"] = statusHistoryData[index].status.name;
+    mass[index]['status'] = statusHistoryData[index].status.name;
```
Comment (ZmicerBoksha) [inline src/components/staff/candidateTrello/candidateTrello.tsx]:
```suggestion
    mass[index].status = statusHistoryData[index].status.name;
```

File: src/components/staff/candidateTrello/candidateTrello.tsx [hunk 13]
```diff
@@ -1,81 +1,63 @@
...
   });
- let notRevived = mass.filter(item =>
-    item.status === "Status name 1");
-  let revived = mass.filter(item =>
-    item.status === "Status name 2"
-  );
+  let notRevived = mass.filter(item => item.status === 'Status name 1');
```
Comment (ZmicerBoksha) [inline src/components/staff/candidateTrello/candidateTrello.tsx]:
```suggestion
  const notRevived = mass.filter(item => item.status === 'Status name 1');
```

File: src/components/staff/candidateTrello/candidateTrello.tsx [hunk 14]
```diff
@@ -1,81 +1,63 @@
...
+  let revived = mass.filter(item => item.status === 'Status name 2');
```
Comment (ZmicerBoksha) [inline src/components/staff/candidateTrello/candidateTrello.tsx]:
```suggestion
  const revived = mass.filter(item => item.status === 'Status name 2');
```

File: src/components/candidates/candidate_card/candidate_interview/candidate_interview.tsx (hunk 4 above)
Comment (vchepurnaya) [reply src/components/candidates/candidate_card/candidate_interview/candidate_interview.tsx]: Corrected

File: src/components/candidates/candidate_card/candidate_progress/candidate_progress.tsx (hunk 5 above)
Comment (vchepurnaya) [reply src/components/candidates/candidate_card/candidate_progress/candidate_progress.tsx]: Corrected

File: src/components/candidates/candidate_card/candidate_progress/candidate_progress.tsx (hunk 6 above)
Comment (vchepurnaya) [reply src/components/candidates/candidate_card/candidate_progress/candidate_progress.tsx]: Corrected

**Review by ZmicerBoksha — APPROVED** [review]

**Review by DmitryPateew — APPROVED** [review]

**Review by ArteGon — APPROVED** [review]

**Pull request merged**